spec/controllers/catalog_controller_spec.rb

```diff
@@ -1,46 +1,81 @@
 require 'rails_helper'
-
 RSpec.describe CatalogController, type: :controller do
-  describe 'POST #add_preserved_object' do
+  describe 'POST #create' do
+    let(:size) { 2342 }
+    let(:version) { 3 }
+    let(:endpoint) { Endpoint.first }
+    let(:druid) { 'bj102hs9687' }
+
     context 'with valid params' do
-      it 'saves the object with the expected field values' do
-        allow(Stanford::StorageServices).to receive(:current_version).and_return(6)
-        allow(subject).to receive(:moab_size).and_return(555)
-        id = 'ab123cd45678'
+      let(:pres_copy) { PreservedCopy.first }
+      let(:pres_obj) { PreservedObject.first }
+
+      before do
+        post :create, params: { druid: druid, incoming_version: version, incoming_size: size, endpoint: endpoint }
+      end
+
+      it 'saves PreservedObject and PreservedCopy in db' do
+        expect(PreservedObject).to exist
+        expect(PreservedCopy).to exist
+      end
+      it 'PreservedCopy and PreservedObject have correct attributes' do
+        expect(pres_copy.endpoint).to eq endpoint
+        expect(pres_copy.version).to eq version
+        expect(pres_copy.size).to eq size
+        expect(pres_obj.druid).to eq druid
+      end
 
-        allow(PreservedObject).to receive(:create!)
-        post :add_preserved_object, params: { id: id }
-        expect(PreservedObject).to have_received(:create!).with(druid: id, size: 555, current_version: 6)
+      it 'response contains create_new_object code ' do
+        code = AuditResults::CREATED_NEW_OBJECT
+        exp_msg = "added object to db as it did not exist"
+        post_response = response.request.env['action_controller.instance'].poh.handler_results.result_array
```

Comment:
Are you saying that in my CatalogController#create method instead of 
```
if poh.handler_results.contains_result_code?(:created_new_object)
      render status: 201, plain: ":created"
    elsif poh.handler_results.contains_result_code?(:db_obj_already_exists)
      render status: 409, plain: ":conflict"
    elsif poh.handler_results.contains_result_code?(:invalid_arguments)
      render status: 406, plain: ":not_acceptable"
    else
      render status: 404, plain: ":not_found"
    end
```

you're saying i should do something like this:
```
if poh.handler_results.contains_result_code?(:created_new_object)
      render status: 201, plain: result_array.json
    elsif poh.handler_results.contains_result_code?(:db_obj_already_exists)
      render status: 409, plain: result_array.json
    elsif poh.handler_results.contains_result_code?(:invalid_arguments)
      render status: 406, plain: result_array.json
    else
      render status: 404, plain: result_array.json
    end

Reply:
Also oops, I should have read your comment below...